ompi/mca/coll/libnbc/libdict/dict_private.h

```diff
@@ -59,8 +60,8 @@ typedef int			 (*icompare_func)	__P((void *, void *itor2));
 # define ASSERT(expr)
 #endif
 
-extern dict_malloc_func _dict_malloc;
-extern dict_free_func _dict_free;
+static dict_malloc_func _dict_malloc;
```

Comment:
Is this header used in multiple TUs? If so we'd end up with multiple copies. If not, we might want to paste these definitions into the single place where it's actually used. Some of the definitions below can be replaced with OPAL variants so most of it unnecessary anyway.

Reply:
This header is included in dict.c and hb_tree.c both in ompi/mca/coll/libnbc/libdict. It looks like these two symbols are only used in dict.c so I guess I can move the definitions there and take them out of this header.

Reply:
It turns out that these symbols are used in MALLOC and FREE macros in dict_private.h and so are referenced in hb_tree.c so can't be static.

If I renamed them, the names would become ompi_mca_coll_libnbc_dict_malloc and ompi_mca_coll_libnbc_dict_free.

If I leave them as _dict_malloc and _dict_free that violates the C standard since symbols starting with '_' are reserved. https://wiki.sei.cmu.edu/confluence/display/c/DCL37-C.+Do+not+declare+or+define+a+reserved+identifier

But that means it's less likely an application is going to use these names, and the code has been this way for a while, so I will leave these as externals with original names.

Reply:
OK, but then they cannot be `static` or one of the copies will not initialized. If they are visible in multiple TUs they should be proper (extern) symbols with proper name spacing. Let's do it right while we're at it :)

Reply:
Yeah, that won't work. I didn't get backing out that change right. I think it's right now.